### PR TITLE
feat: button group component

### PR DESCRIPTION
**Pull request by cody-hoffman**
## Description

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged and published in downstream modules
- [x] The test workflow is passing locally

## Screenshots/GIFs
No scroll
button-group-no-scroll

Scroll
button-group

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 node_modules
-coverage
\ No newline at end of file
+coverage
+storybook-static
```
Comment (twebber90) [inline .gitignore:3]: 👍 just make sure publishing to github pages works as this gets merged in

File: packages/button-group/package.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        "type": "git",
+        "url": "https://github.com/TractorZoom/component-library.git"
+    },
+    "homepage": "https://github.com/TractorZoom/component-library/packages/button-group/README.md",
+    "bugs": {
+        "url": "https://github.com/TractorZoom/component-library/issues"
+    },
+    "keywords": [
+        "material-ui",
+        "button-group"
```
Comment (twebber90) [inline packages/button-group/package.json:21]: 💡 maybe `buttons`, `scrollable`, `scrollable-buttons`

Just some other keywords that came to mine

File: packages/button-group/package.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    "scripts": {
+        "build": "../../node_modules/.bin/babel src --out-dir lib --copy-files",
+        "prepublish": "npm run build"
+    },
+    "peerDependencies": {
+        "@material-ui/core": "^4.10.2",
+        "@tractorzoom/button": "^0.3.0"
+    },
+    "dependencies": {
+        "clsx": "^1.1.1"
```
Comment (twebber90) [inline packages/button-group/package.json:35]: 💡 we should probably add an .npmrc file to each package as well with `save-exact=true`

File: packages/button-group/src/styles.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles((theme) => ({
+    button: {
+        '&:not(:last-child)': {
```
Comment (twebber90) [inline packages/button-group/src/styles.js:5]: 👍

File: stories/button-group/two-buttons.stories.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import ButtonGroup from '../../packages/button-group/src';
+import React from 'react';
+import propsMarkdown from '../utilities/props/button.md';
```
Comment (twebber90) [inline stories/button-group/two-buttons.stories.js:3]: 💡 we should update these stories to have button group prop related documentation (And that markdown file could just mention to look at the Button component for those props)

File: .gitignore (hunk 1 above)
Comment (cody-hoffman) [reply .gitignore:3]: I think it should be fine, this directory didn't exist when I cloned it only existed after I ran a build

File: packages/button-group/package.json (hunk 2 above)
Comment (cody-hoffman) [reply packages/button-group/package.json:21]: :+1:

File: packages/button-group/package.json (hunk 3 above)
Comment (cody-hoffman) [reply packages/button-group/package.json:35]: Is there a reason why we're not using package-locks for the individual packages?

Comment (twebber90) [reply packages/button-group/package.json:35]: This is the first time we've installed a dependency within one of the packages. The other packages just have peer dependencies so there was nothing to install

**Pull request merged**